cmake/faceliftMacros.cmake

```diff
@@ -749,17 +757,41 @@ function(facelift_export_project)
     install(FILES ${CONFIG_DESTINATION_PATH}/${PROJECT_NAME}Config.cmake.installed DESTINATION ${CMAKE_CONFIG_INSTALLATION_PATH} RENAME ${PROJECT_NAME}Config.cmake)
 
     facelift_export_monolithic()
+
 endfunction()
 
+
 function(facelift_export_monolithic)
-    get_property(MONOLITHIC_LIBRARIES GLOBAL PROPERTY MONOLITHIC_LIBRARIES)
-    if(MONOLITHIC_LIBRARIES AND NOT TARGET ${PROJECT_NAME})
-        add_library(${PROJECT_NAME} SHARED)
-        foreach(LIB ${MONOLITHIC_LIBRARIES})
-            target_link_libraries(${PROJECT_NAME} PRIVATE ${LIB})
-        endforeach()
-        install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME}Targets DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+    if (${CMAKE_PROJECT_NAME} STREQUAL ${PROJECT_NAME})  # We only export the main project
+        get_property(MONOLITHIC_LIBRARIES GLOBAL PROPERTY MONOLITHIC_LIBRARIES)
+        if(MONOLITHIC_LIBRARIES AND NOT TARGET ${CMAKE_PROJECT_NAME})
+
+            add_library(${CMAKE_PROJECT_NAME} SHARED)
+
+            foreach(LIB ${MONOLITHIC_LIBRARIES})
+                target_link_libraries(${CMAKE_PROJECT_NAME} PRIVATE ${LIB}__OBJECTS)
+
+                get_property(LINK_LIBRARIES GLOBAL PROPERTY ${LIB}_LINK_LIBRARIES)
+
+                foreach(LINK_LIBRARY ${LINK_LIBRARIES})
+
+                    get_property(IS_MONOLITHIC GLOBAL PROPERTY ${LINK_LIBRARY}_IS_MONOLITHIC SET)
+
+                    if (IS_MONOLITHIC)
+                        target_link_libraries(${LIB} INTERFACE ${LINK_LIBRARY})
+                        target_link_libraries(${LIB}_ INTERFACE ${LINK_LIBRARY}_)
+                    else()
+                        target_link_libraries(${LIB}_ INTERFACE ${LINK_LIBRARY})
+                    endif()
```

Comment:
As agreed some documentation for this is required. It shall be done in near future.